.github/workflows/main.yml

```diff
@@ -0,0 +1,191 @@
+    needs: tests
+    if: startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master'
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu, macos, windows]
+        target: [x86_64, aarch64]
+        manylinux: [auto]
+        include:
+          # Linux
+          - os: ubuntu
+            manylinux: auto
+            target: aarch64
+          - os: ubuntu
+            manylinux: auto
+            target: armv7
+          - os: ubuntu
+            manylinux: auto
+            target: x86_64
+
+          # # musllinux
+          - os: ubuntu
+            target: x86_64
+            manylinux: musllinux_1_1
+          - os: ubuntu
+            target: aarch64
+            manylinux: musllinux_1_1
+
+          # # Windows
+          - os: windows
+            target: x86_64
+          - os: windows
+            target: aarch64
+
+          # MacOs
+          - os: macos
+            target: x86_64
+          - os: macos
+            target: aarch64
+
+    runs-on: ${{ matrix.os }}-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "${{ env.DEFAULT_PYTHON_VERSION }}"
+      - uses: actions/download-artifact@v4
+        with:
+          name: dynamic_version
+          path: src/python_wrapper
+      - run: cat src/python_wrapper/pyproject.toml
+      - run: pip install -U twine
+
+      - name: build sdist
+        if: ${{ matrix.os == 'ubuntu' && matrix.target == 'x86_64' && matrix.manylinux == 'auto' }}
+        uses: PyO3/maturin-action@v1
+        with:
+          command: sdist
+          args: --out ../../dist
+          rust-toolchain: stable
+          working-directory: src/python_wrapper
+
+      - name: build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.target }}
+          manylinux: ${{ matrix.manylinux }}
+          args: --release --out ../../dist --interpreter '3.9 3.10 3.11'
+          rust-toolchain: stable
+          docker-options: -e CI
+          working-directory: src/python_wrapper
+
+      - run: ${{ (matrix.os == 'windows' && 'dir') || 'ls -lh' }} dist/
+
+      - run: twine check --strict dist/*
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: pypi_files_${{ matrix.os }}_${{ matrix.target }}_${{ matrix.manylinux }}
+          path: dist
+
+  cedeartifact_pypi_upload:
+    runs-on: ubuntu-latest
+    needs: build
+    if: startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "${{ env.DEFAULT_PYTHON_VERSION }}"
+      - run: pip install -U twine
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: pypi_files_*
+          merge-multiple: true
+          path: dist
+      - run: du -shc dist/*
+      - uses: aws-actions/configure-aws-credentials@v4
```

Comment:
might be already done but just making sure, did we add permissions to the github oidc role policy ?